api_tests/registries_moderation/test_submissions.py

```diff
@@ -236,3 +312,239 @@ def test_registries_moderation_permission_log(self, app, registration_log_url, r
+        actions_payload_base['data']['attributes']['comment'] = 'Bye bye'
+        actions_payload_base['data']['relationships']['target']['data']['id'] = retract_registration._id
+
+        resp = app.post_json_api(retract_registration_actions_url, actions_payload_base, auth=moderator.auth)
+        assert resp.status_code == 201
+        retract_registration.refresh_from_db()
+        assert retract_registration.moderation_state == RegistrationModerationStates.WITHDRAWN.db_name
+
+    @pytest.mark.enable_quickfiles_creation
+    def test_registries_moderation_post_withdraw_reject(self, app, retract_registration, moderator, retract_registration_actions_url, actions_payload_base, provider):
+        approval_token = retract_registration.sanction.token_for_user(retract_registration.creator, 'approval')
+        retract_registration.sanction.approve(user=retract_registration.creator, token=approval_token)
+
+        actions_payload_base['data']['attributes']['trigger'] = RegistrationModerationTriggers.REJECT_WITHDRAWAL.db_name
+        actions_payload_base['data']['attributes']['comment'] = 'Bye bye'
+        actions_payload_base['data']['relationships']['target']['data']['id'] = retract_registration._id
+
+        resp = app.post_json_api(retract_registration_actions_url, actions_payload_base, auth=moderator.auth)
+        assert resp.status_code == 201
+        retract_registration.refresh_from_db()
+        assert retract_registration.moderation_state == RegistrationModerationStates.ACCEPTED.db_name
+
+    @pytest.mark.enable_quickfiles_creation
+    def test_registries_moderation_post_force_withdraw(self, app, registration, moderator, registration_actions_url, actions_payload_base, provider, reg_creator):
+        registration.require_approval(user=registration.creator)
+
+        actions_payload_base['data']['attributes']['trigger'] = RegistrationModerationTriggers.SUBMIT.db_name
+        actions_payload_base['data']['attributes']['comment'] = 'Submitting registration'
+        actions_payload_base['data']['relationships']['target']['data']['id'] = registration._id
+
+        resp = app.post_json_api(registration_actions_url, actions_payload_base, auth=reg_creator.auth)
+        assert resp.status_code == 201
+        registration.refresh_from_db()
+        assert registration.moderation_state == RegistrationModerationStates.PENDING.db_name
+
+        actions_payload_base['data']['attributes']['trigger'] = RegistrationModerationTriggers.ACCEPT_SUBMISSION.db_name
+        actions_payload_base['data']['attributes']['comment'] = 'Best registration Ive ever seen'
+        actions_payload_base['data']['relationships']['target']['data']['id'] = registration._id
+
+        resp = app.post_json_api(registration_actions_url, actions_payload_base, auth=moderator.auth)
+        assert resp.status_code == 201
+        registration.refresh_from_db()
+        assert registration.moderation_state == RegistrationModerationStates.ACCEPTED.db_name
```

Comment:
You can gloss over this whole rigamarole just by setting registration.is_public if you want.